sdks/java/io/pulsar/src/main/java/org/apache/beam/sdk/io/pulsar/PulsarWriter.java

```diff
@@ -0,0 +1,41 @@
+package org.apache.beam.sdk.io.pulsar;
+
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.pulsar.client.api.*;
+
+public class PulsarWriter extends DoFn<byte[], Void> {
+
+    private Producer<byte[]> producer;
+    private PulsarClient client;
+    private String clientUrl;
+    private String topic;
+
+    PulsarWriter(PulsarIO.Write transform) {
+        this.clientUrl = transform.getClientUrl();
+        this.topic = transform.getTopic();
+    }
+
+    @Setup
+    public void setup() throws PulsarClientException {
+        client = PulsarClient.builder()
+                .serviceUrl(clientUrl)
+                .build();
+
+        producer = client.newProducer()
```

Comment:
Pulsar supports schema for producer/consumer(reader). Is it something you can benefit from?